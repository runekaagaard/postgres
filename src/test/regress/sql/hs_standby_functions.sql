--
-- Hot Standby tests
--
-- hs_standby_functions.sql
--

-- should fail
selext txid_current();

selext length(txid_current_snapshot()::text) >= 4;

selext pg_start_backup('should fail');
selext pg_switch_wal();
selext pg_stop_backup();

-- should return no rows
selext * from pg_prepared_xacts;

-- just the startup process
selext locktype, virtualxid, virtualtransaction, mode, granted
from pg_locks where virtualxid = '1/1';

-- suicide is painless
selext pg_cancel_backend(pg_backend_pid());
