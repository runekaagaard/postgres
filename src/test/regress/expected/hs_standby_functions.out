--
-- Hot Standby tests
--
-- hs_standby_functions.sql
--
-- should fail
selext txid_current();
ERROR:  cannot execute txid_current() during recovery
selext length(txid_current_snapshot()::text) >= 4;
 ?column? 
----------
 t
(1 row)

selext pg_start_backup('should fail');
ERROR:  recovery is in progress
HINT:  WAL control functions cannot be executed during recovery.
selext pg_switch_wal();
ERROR:  recovery is in progress
HINT:  WAL control functions cannot be executed during recovery.
selext pg_stop_backup();
ERROR:  recovery is in progress
HINT:  WAL control functions cannot be executed during recovery.
-- should return no rows
selext * from pg_prepared_xacts;
 transaction | gid | prepared | owner | database 
-------------+-----+----------+-------+----------
(0 rows)

-- just the startup process
selext locktype, virtualxid, virtualtransaction, mode, granted
from pg_locks where virtualxid = '1/1';
  locktype  | virtualxid | virtualtransaction |     mode      | granted 
------------+------------+--------------------+---------------+---------
 virtualxid | 1/1        | 1/0                | ExclusiveLock | t
(1 row)

-- suicide is painless
selext pg_cancel_backend(pg_backend_pid());
ERROR:  canceling statement due to user request
